File: python/docmost/services/search_api.py
"""Generated API client."""
from __future__ import annotations

from typing import Any, Optional

import httpx

from docmost.http_client import DocmostHttpClient
from docmost.types import ApiResponse
from docmost.models import SearchDto, SearchShareDto, SearchSuggestionDto


class SearchApi:
    """Search API."""

    def __init__(self, http: DocmostHttpClient) -> None:
        self._http = http

    def page_search(self, request: SearchDto) -> ApiResponse[Any]:
        return self._http.post("search", body=request.model_dump(by_alias=True, exclude_none=True))

    def search_suggestions(self, request: SearchSuggestionDto) -> ApiResponse[Any]:
        return self._http.post("search/suggest", body=request.model_dump(by_alias=True, exclude_none=True))

    def search_share(self, request: SearchShareDto) -> ApiResponse[Any]:
        return self._http.post("search/share-search", body=request.model_dump(by_alias=True, exclude_none=True))

File: go/docmost/services/search_api.go
package services

import (
	"context"

	"github.com/ggpo1/docmost-sdk/go/internal/httpclient"
	"github.com/ggpo1/docmost-sdk/go/docmost/models"
)

// SearchAPI implements Search API.
type SearchAPI struct {
	http *httpclient.HTTPClient
}

func NewSearchAPI(http *httpclient.HTTPClient) *SearchAPI {
	return &SearchAPI{http: http}
}

func (a *SearchAPI) PageSearch(ctx context.Context, request *models.SearchDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "search", request)
}

func (a *SearchAPI) SearchSuggestions(ctx context.Context, request *models.SearchSuggestionDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "search/suggest", request)
}

func (a *SearchAPI) SearchShare(ctx context.Context, request *models.SearchShareDto) (*httpclient.ApiResponse, error) {
	return a.http.Post(ctx, "search/share-search", request)
}

